src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -155,154 +186,253 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
-
-            var isHls = streamInfo.url.toLowerCase().indexOf('.m3u8') !== -1;
+        /**
+         * @private
+         */
+        updateVideoUrl(streamInfo) {
+            const isHls = streamInfo.url.toLowerCase().includes('.m3u8');
 
-            var mediaSource = streamInfo.mediaSource;
-            var item = streamInfo.item;
+            const mediaSource = streamInfo.mediaSource;
+            const item = streamInfo.item;
 
             // Huge hack alert. Safari doesn't seem to like if the segments aren't available right away when playback starts
             // This will start the transcoding process before actually feeding the video url into the player
             // Edit: Also seeing stalls from hls.js
             if (mediaSource && item && !mediaSource.RunTimeTicks && isHls && streamInfo.playMethod === 'Transcode' && (browser.iOS || browser.osx)) {
 
-                var hlsPlaylistUrl = streamInfo.url.replace('master.m3u8', 'live.m3u8');
+                const hlsPlaylistUrl = streamInfo.url.replace('master.m3u8', 'live.m3u8');
 
                 loading.show();
 
-                console.debug('prefetching hls playlist: ' + hlsPlaylistUrl);
+                console.debug(`prefetching hls playlist: ${hlsPlaylistUrl}`);
 
                 return connectionManager.getApiClient(item.ServerId).ajax({
 
                     type: 'GET',
                     url: hlsPlaylistUrl
 
                 }).then(function () {
-
-                    console.debug('completed prefetching hls playlist: ' + hlsPlaylistUrl);
+                    console.debug(`completed prefetching hls playlist: ${hlsPlaylistUrl}`);
 
                     loading.hide();
                     streamInfo.url = hlsPlaylistUrl;
-
-                    return Promise.resolve();
-
                 }, function () {
-
-                    console.error('error prefetching hls playlist: ' + hlsPlaylistUrl);
+                    console.error(`error prefetching hls playlist: ${hlsPlaylistUrl}`);
 
                     loading.hide();
-                    return Promise.resolve();
                 });
 
             } else {
                 return Promise.resolve();
             }
         }
 
-        self.play = function (options) {
-            self._started = false;
-            self._timeUpdated = false;
-
-            self._currentTime = null;
+        play(options) {
+            this.#started = false;
+            this.#timeUpdated = false;
 
-            self.resetSubtitleOffset();
+            this.#currentTime = null;
 
-            return createMediaElement(options).then(function (elem) {
+            this.resetSubtitleOffset();
 
-                return updateVideoUrl(options).then(function () {
-                    return setCurrentSrc(elem, options);
+            return this.createMediaElement(options).then(elem => {
+                return this.updateVideoUrl(options).then(() => {
+                    return this.setCurrentSrc(elem, options);
```

Comment:
This will fix forced transcoding.
```suggestion
                    return this.setCurrentSrc(this.#mediaElement, options);
```

Reply:
This is wrong. I've found the right one :crossed_fingers: